### PR TITLE
Merge UI logic into favourites

### PR DESCRIPTION
**Pull request by coreylevine**
Features added:
1. Favourite button is added on each card.
2. When you click on the favourites button, it gets filled.
3. There is a favourites deck icon in miscellaneous tab

### CODE REVIEW

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java [hunk 1]
```diff
@@ -362,6 +365,9 @@ public void onClick(DialogInterface arg0, int arg1){
         if(v == aboutButton){
             aboutUtil.createAboutDialog();
         }
+        if(v == favoriteButton){
+
+        }
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java]: Why do you have this if statement?

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java [hunk 2]
```diff
@@ -86,6 +86,7 @@
     private View donateButton;
     private View helpButton;
     private View aboutButton;
+    private View favoriteButton;
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java]: In the other favourite branch we write favorite with the following spelling: "favourite". Please change your spelling to that.

**Comment by j-suguma** [comment]
CI build for favUI:
image

tests:
image

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java (hunk 1 above)
Comment (j-suguma) [reply app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java]: That is just there temporarily, it is where the backend logic will be placed

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java (hunk 2 above)
Comment (j-suguma) [reply app/src/main/java/org/liberty/android/fantastischmemo/ui/MiscTabFragment.java]: Okay ill take this into account and change the spelling so it matches with the rest of the code when I work on this new merged branch

**Comment by j-suguma** [comment]
CI build for favFields:
image

tests:
image

**Pull request merged**